composer/loggers/cometml_logger.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Log to `Comet <https://www.comet.com/docs/v2/>`_."""
+
+from __future__ import annotations
+
+from typing import Any, Dict, Optional
+
+from composer.core.state import State
+from composer.loggers.logger import Logger
+from composer.loggers.logger_destination import LoggerDestination
+from composer.utils import dist
+from composer.utils.import_helpers import MissingConditionalImportError
+
+__all__ = ['CometMLLogger']
+
+
+class CometMLLogger(LoggerDestination):
+    """Log to `Comet <https://www.comet.com/docs/v2/>`_.
+
+    Args:
+        workspace (str, optional): The name of the workspace which contains the project
+            you want to attach your experiment to. If nothing specified will default to your
+            default workspace as configured in your comet account settings.
+        project_name (str, optional): The name of the project to categorize your experiment in.
+            A new project with this name will be created under the Comet workspace if one
+            with this name does not exist. If no project name specified, the experiment will go
+            under Uncategorized Experiments.
+        log_code (bool): Whether to log your code in your experiment (default: ``False``).
+        log_graph (bool): Whether to log your computational graph in your experiment
+            (default: ``False``).
+        name (str, optional): The name of your experiment. If not specified, it will be set
+            to :attr:`.State.run_name`.
+        rank_zero_only (bool, optional): Whether to log only on the rank-zero process.
+            (default: ``False``).
+        exp_kwargs (Dict[str, Any], optional): Any additional kwargs to
+            comet_ml.Experiment(see
+            `Comet documentation <https://www.comet.com/docs/v2/api-and-sdk/python-sdk/reference/Experiment/>`_).
+    """
+
+    def __init__(
+        self,
+        workspace: Optional[str] = None,
+        project_name: Optional[str] = None,
+        log_code: bool = False,
+        log_graph: bool = False,
+        name: Optional[str] = None,
+        rank_zero_only: bool = True,
+        exp_kwargs: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        try:
+            from comet_ml import Experiment
+        except ImportError as e:
+            raise MissingConditionalImportError(extra_deps_group='comet_ml',
+                                                conda_package='comet_ml',
+                                                conda_channel='conda-forge') from e
```

Comment:
Note: composer currently doesn't support `conda`. What exactly does this do?

Reply:
yeah it's a weird practice I'm following from our [style guide](https://github.com/mosaicml/composer/blob/dev/STYLE_GUIDE.md#51-external-dependencies)